openhcl/openhcl_boot/src/hypercall.rs

```diff
@@ -386,6 +416,96 @@ impl HvCall {
     }
 }
 
+// TDX specific hypercall methods
+#[cfg(target_arch = "x86_64")]
+impl HvCall {
+    /// Initialize HvCall for use in a TDX CVM
+    pub fn initialize_tdx(&mut self, tdx_io_page: TdxHypercallPage) {
+        assert!(!self.initialized);
+        self.initialized = true;
+        self.vtl = Vtl::Vtl2;
+        self.tdx_io_page = Some(tdx_io_page);
+
+        // TODO: revisit os id value. For now, use 1 (which is what UEFI does)
+        let guest_os_id = hvdef::hypercall::HvGuestOsMicrosoft::new().with_os_id(1);
+
+        crate::arch::tdx::initialize_hypercalls(
+            guest_os_id.into(),
+            self.tdx_io_page.as_ref().unwrap(),
+        );
+    }
+
+    /// Uninitialize HvCall for TDX, unshare IO pages with the hypervisor
+    pub fn uninitialize_tdx(&mut self) {
+        assert!(self.tdx_io_page.is_some());
+        crate::arch::tdx::uninitialize_hypercalls(
+            self.tdx_io_page
+                .take()
+                .expect("an initialized instance of HvCall on TDX must have an io page"),
+        );
+    }
+
+    /// Hypercall to enable VTL2 on a TDX VP
+    pub fn tdx_enable_vp_vtl2(&mut self, vp_index: u32) -> Result<(), hvdef::HvError> {
+        let header = hvdef::hypercall::EnableVpVtlX64 {
+            partition_id: hvdef::HV_PARTITION_ID_SELF,
+            vp_index,
+            // The VTL value here is just a u8 and not the otherwise usual
+            // HvInputVtl value.
+            target_vtl: Vtl::Vtl2.into(),
+            reserved: [0; 3],
+            // Context must be zeroed for the VP VTL startup hypercall on TDX
+            vp_vtl_context: zerocopy::FromZeros::new_zeroed(),
+        };
+        assert!(self.initialized);
+        assert!(self.tdx_io_page.is_some());
```

Comment:
i think it's a bit clearer to also remove this assert, then replace the `unwrap` below with `.expects("should only be called on tdx")`

Reply:
same for `tdx_start_vp` as well